app/models/upload.rb

```diff
@@ -544,6 +509,25 @@ def self.migrate_to_new_scheme(limit: nil)
     problems
   end
 
+  def self.extract_upload_ids(raw)
+    return [] if raw.blank?
+
+    urls = []
+    sha1s = []
+
+    InlineUploads.url_regexps.each do |regexp|
+      raw.scan(regexp).each do |match|
+        urls << match[0]
+        if match[1].present?
+          sha1s << match[1] if match[1] =~ /^[0-9a-fA-f]+$/
```

Comment:
In [Ruby](https://ruby-doc.org/core-3.1.1/Regexp.html#class-Regexp-label-Character+Classes), you can shorten this regexp using the `\h` character class

```suggestion
          sha1s << match[1] if match[1] =~ /^\h+$/
```